mono/tests/unhandled-exception-6.cs

```diff
@@ -2,40 +2,32 @@
 using System.Diagnostics;
 using System.Threading;
 using System.Threading.Tasks;
+using System.Runtime.Remoting.Messaging;
 
 class CustomException : Exception
 {
 }
 
-class CustomException2 : Exception
-{
-}
-
 class Driver
 {
 	/* expected exit code: 0 */
 	static void Main (string[] args)
 	{
-		ManualResetEvent mre = new ManualResetEvent (false);
-		ManualResetEvent mre2 = new ManualResetEvent (false);
+		var action = new Action (Delegate);
+		var ares = action.BeginInvoke (Callback, null);
 
-		var a = new Action (() => { try { throw new CustomException (); } finally { mre.Set (); } });
-		var ares = a.BeginInvoke (_ => { mre2.Set (); throw new CustomException2 (); }, null);
+		Thread.Sleep (5000);
```

Comment:
Their should be a unhandled exception in the ThreadPool, so if we ever hit that 5 seconds threshold, it means we fail the test. So we can be optimistic and say that we should never wait 5 seconds :)